src/Datadog.Trace/Headers/NameValueHeadersCollection.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Datadog.Trace.Headers
 {
-    internal class NameValueHeadersCollection : IHeadersCollection
+    internal readonly struct NameValueHeadersCollection : IHeadersCollection
```

Comment:
Are these changes from `class` to `struct` breaking if called from an older version of `Datadog.Trace.ClrProfiler.Managed`?

Reply:
Definitely, returning an interface reference (boxing if steuct) vs returning a value. There’s a couple breaking changes related to this, but reduces heap allocations.

Reply:
I think we should prioritize stability over performance / resource usage, at least until we resolve the version mismatch issues.

Reply:
then maybe we should push back these improvements until we resolve the compatibilities issues we have. I'll resolve and handle all the requested changes but leave the PR with the `no-merge` flag.

Reply:
thanks!